src/client/Client.cc

```diff
@@ -376,13 +376,7 @@ void Client::tear_down_cache()
   // close root ino
   ceph_assert(inode_map.size() <= 1 + root_parents.size());
   if (root && inode_map.size() == 1 + root_parents.size()) {
-    delete root;
-    root = 0;
-    root_ancestor = 0;
-    while (!root_parents.empty())
-      root_parents.erase(root_parents.begin());
-    inode_map.clear();
-    _reset_faked_inos();
+    put_inode(root, 0);
```

Comment:
throw in some asserts to make sure `root_parents` is empty?

Reply:
Here the `root_parent` is allowed not to be empty and it will be erased and cleaned later in `Client::_put_inode()`.